{
    "CVE_data_meta": {
        "ASSIGNER": "security-advisories@github.com",
        "ID": "CVE-2022-29196",
        "STATE": "PUBLIC",
        "TITLE": "Missing validation causes denial of service in TensorFlow via `Conv3DBackpropFilterV2`"
    },
    "affects": {
        "vendor": {
            "vendor_data": [
                {
                    "product": {
                        "product_data": [
                            {
                                "product_name": "tensorflow",
                                "version": {
                                    "version_data": [
                                        {
                                            "version_value": "< 2.6.4"
                                        },
                                        {
                                            "version_value": ">= 2.7.0rc0, < 2.7.2"
                                        },
                                        {
                                            "version_value": ">= 2.8.0rc0, < 2.8.1"
                                        },
                                        {
                                            "version_value": ">= 2.9.0rc0, < 2.9.0"
                                        }
                                    ]
                                }
                            }
                        ]
                    },
                    "vendor_name": "tensorflow"
                }
            ]
        }
    },
    "data_format": "MITRE",
    "data_type": "CVE",
    "data_version": "4.0",
    "description": {
        "description_data": [
            {
                "lang": "eng",
                "value": "TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.Conv3DBackpropFilterV2` does not fully validate the input arguments. This results in a `CHECK`-failure which can be used to trigger a denial of service attack. The code does not validate that the `filter_sizes` argument is a vector. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue."
            }
        ]
    },
    "impact": {
        "cvss": {
            "attackComplexity": "LOW",
            "attackVector": "LOCAL",
            "availabilityImpact": "HIGH",
            "baseScore": 5.5,
            "baseSeverity": "MEDIUM",
            "confidentialityImpact": "NONE",
            "integrityImpact": "NONE",
            "privilegesRequired": "LOW",
            "scope": "UNCHANGED",
            "userInteraction": "NONE",
            "vectorString": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
            "version": "3.1"
        }
    },
    "problemtype": {
        "problemtype_data": [
            {
                "description": [
                    {
                        "lang": "eng",
                        "value": "CWE-20: Improper Input Validation"
                    }
                ]
            }
        ]
    },
    "references": {
        "reference_data": [
            {
                "name": "https://github.com/tensorflow/tensorflow/releases/tag/v2.6.4",
                "refsource": "MISC",
                "url": "https://github.com/tensorflow/tensorflow/releases/tag/v2.6.4"
            },
            {
                "name": "https://github.com/tensorflow/tensorflow/releases/tag/v2.7.2",
                "refsource": "MISC",
                "url": "https://github.com/tensorflow/tensorflow/releases/tag/v2.7.2"
            },
            {
                "name": "https://github.com/tensorflow/tensorflow/releases/tag/v2.8.1",
                "refsource": "MISC",
                "url": "https://github.com/tensorflow/tensorflow/releases/tag/v2.8.1"
            },
            {
                "name": "https://github.com/tensorflow/tensorflow/releases/tag/v2.9.0",
                "refsource": "MISC",
                "url": "https://github.com/tensorflow/tensorflow/releases/tag/v2.9.0"
            },
            {
                "name": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-5v77-j66x-4c4g",
                "refsource": "CONFIRM",
                "url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-5v77-j66x-4c4g"
            },
            {
                "name": "https://github.com/tensorflow/tensorflow/commit/174c5096f303d5be7ed2ca2662b08371bff4ab88",
                "refsource": "MISC",
                "url": "https://github.com/tensorflow/tensorflow/commit/174c5096f303d5be7ed2ca2662b08371bff4ab88"
            },
            {
                "name": "https://github.com/tensorflow/tensorflow/blob/f3b9bf4c3c0597563b289c0512e98d4ce81f886e/tensorflow/core/kernels/conv_grad_ops_3d.cc",
                "refsource": "MISC",
                "url": "https://github.com/tensorflow/tensorflow/blob/f3b9bf4c3c0597563b289c0512e98d4ce81f886e/tensorflow/core/kernels/conv_grad_ops_3d.cc"
            }
        ]
    },
    "source": {
        "advisory": "GHSA-5v77-j66x-4c4g",
        "discovery": "UNKNOWN"
    }
}